src/beaconer.rs

```diff
@@ -25,6 +25,18 @@ pub enum Message {
     RegionParamsChanged(RegionParams),
 }
 
+#[derive(Debug)]
+pub struct BeaconReq {
+    pub beacon: beacon::Beacon,
+    pub response: sync::ResponseSender<Result<BeaconResp>>,
+}
+
+#[derive(Debug)]
+pub struct BeaconResp {
+    pub powe: Option<i32>,
```

Comment:
Afaict these are never set to None in the responder, so why declare them as Option here?